### DISCUSSION TITLE
AWS IMDS v2 support

### DISCUSSION THREAD

**Discussion by geanboegman** [ruby/rubygems#5757] [Q&A]
Is there any plan to update the [s3_uri_signer](https://github.com/rubygems/rubygems/blob/4802a99bbf43a6600eeda4344d9d5460e660d6ae/lib/rubygems/s3_uri_signer.rb#L152) to use session tokens as defined in the more secure [IMDSv2](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) API?  If so please can someone point me at the tasks as I could not find one.

**Comment by simi** [comment]
AFAIK there are no plans for this feature.

**Reply by geanboegman** [reply]
I know the IT industry in general is moving towards IMDSv2 everywhere, I guess this is not a critical feature but the update should be relatively trivial.  Or will this feature go out of support over time (asking as I do not have insight into the roadmap for this feature)

**Reply by simi** [reply]
Would you like to contribute this? I can help if needed.

**Reply by geanboegman** [reply]
I'll see, possibly.  I've created a ticket for it (https://github.com/rubygems/rubygems/issues/5763)

**Comment by deivid-rodriguez** [comment]
This turned into a feature request, so closing this discussion in favor of that.
